src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -828,12 +828,14 @@ public Edge(Guid startId, Guid endId, double startX, double startY, double endX,
             OwnerGraph = ownerGraph;
 
             StartNode = OwnerGraph.FindNode(startId);
+            if(StartNode is null) { return; }
             if (StartNode != null)
             {
                 StartNode.RightEdges.Add(this);
             }
 
             EndNode = OwnerGraph.FindNode(endId);
+            if(EndNode is null) { return; }
             if (EndNode != null)
```

Comment:
this is a little strange? why have you added the return if its null?

Reply:
Code is a little redundant. Should remove` if (EndNode != null)` and keep BAIL QUICK return pattern.

Reply:
Done.